tests/utils.go

```diff
@@ -2764,6 +2764,22 @@ func WaitForSuccessfulVMIStart(vmi runtime.Object) string {
 	return waitForVMIStart(vmi, 360, false)
 }
 
+func WaitUntilVMIReadyAsync(vmi *v1.VirtualMachineInstance, expecterFactory VMIExpecterFactory) func() *v1.VirtualMachineInstance {
+	var (
+		wg       sync.WaitGroup
+		readyVMI *v1.VirtualMachineInstance
+	)
+	wg.Add(1)
+	go func() {
+		readyVMI = WaitUntilVMIReady(vmi, expecterFactory)
+	}()
+
+	return func() *v1.VirtualMachineInstance {
+		wg.Wait()
```

Comment:
better to add some bounded waiting ? (timer)
also where you decrease the WG (Done)

Reply:
Ahh you right I forgot the `defer`, about the timeout the function that calls the go routine is implemented with Eventually at the end so we have an implicit timeout mechanism there.

Reply:
@qinqon @oshoval is right. I guess the overlap is not that big here, but if the test fails we may have go-routines left over for some time. That can c ause confusion. You will have to add an additional stop and done channel. In case of an error, we need to stop the watchers, since one of the watchers may still continue.